lighthouse-core/audits/seo/robots-txt.js

```diff
@@ -198,6 +198,11 @@ class RobotsTxt extends Audit {
       };
     }
 
+    // If status is good, content must be not null.
+    if (content === null) {
+      throw new Error(`Status ${status} was valid, but content was null`);
+    }
```

Comment:
@kdzwinel had to add this to keep the type checker happy even though it appears to not be possible. I couldn't think of a good way around this short of adding another property that lets you demonstrate to the type checker that a null `content` can't happen in certain cases, e.g.
`{error: false, status: number, content: string}|{error: true, status: number|null, content: string|null}`
...or we can just test like here for `// can't ever happen` :)